From 4a6b02282632f0b4e88a85f26266f7674e0ce288 Mon Sep 17 00:00:00 2001
From: Namjae Jeon <namjae.jeon@samsung.com>
Date: Wed, 26 May 2021 15:28:09 +0900
Subject: [PATCH] cifsd: simplify error handling in ksmbd_auth_ntlm()
Git-commit: 4a6b02282632f0b4e88a85f26266f7674e0ce288
Patch-mainline: v5.15-rc1
References: bsn#12
Modified-by-SEL: No


simplify error handling in ksmbd_auth_ntlm().

Reviewed-by: Dan Carpenter <dan.carpenter@oracle.com>
Signed-off-by: Namjae Jeon <namjae.jeon@samsung.com>
Signed-off-by: Steve French <stfrench@microsoft.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 fs/cifsd/auth.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/fs/cifsd/auth.c b/fs/cifsd/auth.c
index 092db15e4234..7771429f55a4 100644
--- a/fs/cifsd/auth.c
+++ b/fs/cifsd/auth.c
@@ -351,12 +351,11 @@ int ksmbd_auth_ntlm(struct ksmbd_session *sess, char *pw_buf)
 
 	if (strncmp(pw_buf, key, CIFS_AUTH_RESP_SIZE) != 0) {
 		ksmbd_debug(AUTH, "ntlmv1 authentication failed\n");
-		rc = -EINVAL;
-	} else {
-		ksmbd_debug(AUTH, "ntlmv1 authentication pass\n");
+		return -EINVAL;
 	}
 
-	return rc;
+	ksmbd_debug(AUTH, "ntlmv1 authentication pass\n");
+	return 0;
 }
 
 /**
-- 
2.33.0

